pkgs/by-name/nv/nvidia-container-toolkit/package.nix

```diff
@@ -56,6 +56,12 @@ buildGoModule rec {
     ./0001-Add-dlopen-discoverer.patch
   ];
 
+  subPackages = [
+    "cmd/nvidia-ctk"
+    "cmd/nvidia-container-runtime"
+    "cmd/nvidia-container-runtime-hook"
+  ];
+
```

Comment:
I'm stupid, what does this do?

Reply:
`nvidia-container-toolkit` has a lot of binaries in its tree. Some of them are under https://github.com/NVIDIA/nvidia-container-toolkit/tree/main/cmd. Others are under https://github.com/NVIDIA/nvidia-container-toolkit/tree/main/tools/container.

The last one is the one causing the shadowing, but in reality the main issue is that some of those despite being named `docker` or `crio` are not meant to shadow the original binaries, but just named like that to set up the corresponding runtime configuration.

I believe we don't need to expose that kind of tooling to the user, and that NixOS will properly configure the given runtime with regular NixOS options. This is an implementation detail in my opinion.

So with `subPackages` what we are telling `buildGoModule` is: instead of building all the binaries that you can find in this tree structure, just build `cmd/nvidia-ctk`, `cmd/nvidia-container-runtime` and `cmd/nvidia-container-runtime-hook`.

To be honest, I would even just try with `cmd/nvidia-ctk` and the start adding more binaries as we see are missing, but this was a way to start.

Reply:
This is the list of binaries produced by `nix-build -A nvidia-docker` before this change:

```
❯ tree result/bin
result/bin
├── containerd -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/containerd
├── crio -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/crio
├── docker -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/docker
├── nvidia-container-cli -> /nix/store/570j23jms28cjb603a7hy695nmibfi0n-libnvidia-container-1.9.0/bin/nvidia-container-cli
├── nvidia-container-runtime -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-runtime
├── nvidia-container-runtime.cdi -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-runtime.cdi
├── nvidia-container-runtime-hook -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-runtime-hook
├── nvidia-container-runtime.legacy -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-runtime.legacy
├── nvidia-container-toolkit -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-toolkit
├── nvidia-ctk -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-ctk
├── nvidia-docker -> /nix/store/lc86iqnadxhc54n34n4g9rd94li39ls8-nvidia-docker-2.5.0/bin/nvidia-docker
├── nvidia-toolkit -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-toolkit
└── toolkit -> /nix/store/9933i2f5wnj6f994isn242vdnbzdry62-container-toolkit-container-toolkit-1.15.0-rc.3/bin/toolkit
```

And after this change:

```
❯ tree result/bin
result/bin
├── nvidia-container-cli -> /nix/store/570j23jms28cjb603a7hy695nmibfi0n-libnvidia-container-1.9.0/bin/nvidia-container-cli
├── nvidia-container-runtime -> /nix/store/iw49hbrnnxg2pf1ynha35gn96mgpg4xx-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-runtime
├── nvidia-container-runtime-hook -> /nix/store/iw49hbrnnxg2pf1ynha35gn96mgpg4xx-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-runtime-hook
├── nvidia-container-toolkit -> /nix/store/iw49hbrnnxg2pf1ynha35gn96mgpg4xx-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-container-toolkit
├── nvidia-ctk -> /nix/store/iw49hbrnnxg2pf1ynha35gn96mgpg4xx-container-toolkit-container-toolkit-1.15.0-rc.3/bin/nvidia-ctk
└── nvidia-docker -> /nix/store/lc86iqnadxhc54n34n4g9rd94li39ls8-nvidia-docker-2.5.0/bin/nvidia-docker
```

Reply:
So instead of moving runtime wrappers to a separate output or prefix, this removes the wrappers entirely. I've no idea if the wrappers still have any users, I think we should go with the moving instead

Reply:
Please, note that the wrapper is `nvidia-docker`. The rest are just helper binaries that configure the runtime after they are called: tools that perform imperative changes on the host to set up the runtime wrapper, but not the wrappers themselves.

Reply:
Marking this PR as draft until I have double checked everything I want.